### ISSUE TITLE
TiddlyDesktop v0.0.4 64bits linux version doesn't launch

### ISSUE THREAD

**Issue by Spangenhelm** [Jermolene/TiddlyDesktop#53]
Apparently the nw app does not launch at all (permission is set to executable) when trying from the file manager (nothing happen, no error displayed), unfortunatly i don't know how to launch it from the terminal, i have tried "nw" and "exec nw" with the right path but the cli return this: "bash: nw command not found" anyway i don't have this problem with 0.0.3 which works fine.

**Comment by isaacrg** [comment]
Use `./nw`, while in the same folder as the app. Does that work?

**Comment by Jermolene** [comment]
Thanks @isaacrg. @Spangenhelm if there's a difference in the launch behaviour from 0.0.3 to 0.0.4 then I think it must be something that's changed in node-webkit itself (or nw.js as it's now called).

**Comment by Spangenhelm** [comment]
Thanks fo the tip @isaacrg it allowed me to know what was going on.

@Jermolene you are right the problem comes from [libudev from nw.js] (https://github.com/nwjs/nw.js/wiki/The-solution-of-lacking-libudev.so.0) but after correcting this problem now it shows me this error:

`[8704:0207/114242:ERROR:browser_main_loop.cc(162)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.`

**Comment by Jermolene** [comment]
@Spangenhelm the referenced URL barely makes sense to me, I'm afraid. Perhaps @isaacrg can help interpret it

**Comment by braykov** [comment]
@Jermolene  I got the same error message. But nw opens successfully.
However, same as on Windows, wikis act slower than before. For example after I resize an individual wiki window, it shows a black rectangle and takes 3-4 sec. to refresh and resize the contents inside. Clicking links inside the wiki, opening tiddlers, etc. is also slower.
Then the biggest problem  (again, same as on Windows) after closing nw, it appears to be still working (I don't get a prompt) so then it cannot open again unless I kill the process.

**Comment by Jermolene** [comment]
Hi @braykov #52 covers the window resizing and phantom process issues, but it may be worth starting an issue for the general wiki performance regression.

**Comment by braykov** [comment]
@Jermolene I would prefer to wait for next version or at least to have a properly running nw. Once I can freely start and stop TiddlyWiki I will start using it on my laptop with Ubuntu 14.10 and at work on Windows 7. I can then compare the behavior of both OS + with 0.0.3.

**Comment by Spangenhelm** [comment]
Hi @Jermolene, i may have found something with this link: https://github.com/nwjs/nw.js/issues/2699

Apparently this is just an error, below is the message of the creator of nw:
>This can be ignored, since we don't use renderer sandbox in node-webkit. I'll remove this message.

Still, not only this message is displayed but most importantly nw is not running at all!

**Comment by Jermolene** [comment]
@Spangenhelm I'm not actually compiling the Linux build (or any of them). I just download the compiled binaries from the nw.js releases page and then splice in our HTML file and other content.

**Comment by Jermolene** [comment]
Hi @Spangenhelm of course, TiddlyDesktop could do with an upgrade to the latest nw.js.

**Comment by Spangenhelm** [comment]
Hi @Jermolene 

I have just tried the last version of Nw on Linux 64: "nwjs-v0.12.1-linux-x64", it works fine but.. without Tiddlywiki! __Same thing with the nw of your last release (TiddlyDesktop v0.0.4).__
To run it "alone" i had to rename the `package.json` file (eg: by adding ".old" to it) to prevent nw from using it and trying to load Tw. Like this no problem (start and stops without hicups, nothing stays in the background processes when closed), 
But if i try with it then it won't even load, at all. (nothing happens exept the famous SUID sandbox warning in the terminal)

Well, when in the `package.json` file, if i modify the "name" entry for example (add or remove any letter) then it starts fine but there is something in Tw that prevent it from closing properly (there are 4 processes that stays in the background after closing, as @braykov mentionned)

So to summarize: To me the problem come from tw and not from nw.js (except for the warning message) or, does the issue simply come from a missing "adjustment" to make tw compatible with the new nw version?

+++ Spangenhelm +++

**Comment by Jermolene** [comment]
Hi @Spangenhelm I haven't investigated if any changes to TiddlyDesktop are needed in order to work with nw.js 0.12.1, but it's entirely possible that there.

I think there are two main, independent issues in the frame:

* Quitting TiddlyDesktop doesn't work properly, leaving phantom processes behind. This is an outstanding bug that needs fixing
* The warning from nw.js that was reported by you on Feb 7th - https://github.com/Jermolene/TiddlyDesktop/issues/53#issuecomment-73359574. I'm expecting this to be fixed by the new version

**Comment by Spangenhelm** [comment]
@Jermolene the warning is still present in the latest version i have tried and the issue is still open over there so don't expect the correction too soon unfortunatly..

**Comment by Jermolene** [comment]
Thanks @Spangenhelm

**Comment by guraltsev** [comment]
I am also observing the same thing: TiddlyWiki doesn't exit under Linux when closing the last window

**Comment by Jermolene** [comment]
I think the phantom process issue is fixed in 56892f66d5860974a47022d9e77df552087501a8. I'll issue v0.0.5 soon, but if anybody has the chance to try their own build please let me know if it works for you.

**Comment by Jermolene** [comment]
Closing as fixed in v0.0.6
